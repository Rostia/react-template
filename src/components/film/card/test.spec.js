import React from 'react';
import FilmCard from 'components/film/card/index';
import { BrowserRouter as Router } from 'react-router-dom';

const mockFilm = {
  id: 1,
  title: 'Transformer',
  release_date: '2019-06-26',
  genres: ['Action', 'Adventure'],
  poster_path: 'https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg',
};

describe('FilmCard component', () => {
  it("should render FilmCard component", () => {
    const component = render(<Router><FilmCard film={mockFilm} /></Router>);
    expect(component).toMatchSnapshot();
  });
});
