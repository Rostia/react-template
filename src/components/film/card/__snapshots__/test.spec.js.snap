// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard component should render FilmCard component 1`] = `
<div
  class="cardContained"
>
  <a
    class="card"
    href="/film/1"
  >
    <img
      alt="Film"
      class="cardImage"
      src="https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg"
    />
    <div
      class="row"
    >
      <h3
        class="cardTitle"
      >
        Transformer
      </h3>
      <span
        class="cardYear"
      >
        2019-06-26
      </span>
    </div>
    <span
      class="cardGanre"
    >
      Action, Adventure
    </span>
  </a>
  <button
    class="actionButton actionButton"
    type="button"
  >
    <span
      class="dotes"
    >
      ...
    </span>
  </button>
</div>
`;
